### ISSUE TITLE
PKCS12_parse leaves errors on stack [3.0.6]

### ISSUE THREAD

**Issue by reaperhulk** [openssl/openssl#19389]
OpenSSL 3.0.6 has a regression where successful parse of a PKCS12 via `PKCS12_parse` leaves errors on the stack. This appears to be due to changes to error handling in functions that `PKCS12_parse` calls.

**Comment by alex** [comment]
It's because of https://github.com/openssl/openssl/blob/openssl-3.0/crypto/pkcs12/p12_kiss.c#L92

shift was changed to put an error on the stack if you shfit from an enpty stack

this regression is trivially observed by running the pyca/cryptography test suite

**Comment by t8m** [comment]
The regression is caused by #18918

**Comment by paulidale** [comment]
@DDvO ??

**Comment by paulidale** [comment]
@DDvO ?

**Comment by slontis** [comment]
@reaperhulk - Does this mean that the external tests that we have for pyca are older than the ones we currently have? Any chance this could be updated?

**Comment by alex** [comment]
Your external tests are not properly configured. Looking at a recent job from the 3.0 branch: https://github.com/openssl/openssl/actions/runs/3226931046/jobs/5281077558

> OpenSSL: OpenSSL 1.1.1f  31 Mar 2020

You are linking pyca/cryptography against a system OpenSSL, not the OpenSSL you have built.

It's also the case that your copy of pyca/cryptography is a year old, but that's not relevant here.

**Comment by slontis** [comment]
Has this been broken since it got put in (6 years ago)

**Comment by slontis** [comment]
@alex Isnt the -L option supposed to handle this?
i.e.
````
CFLAGS=-I/home/runner/work/openssl/openssl/test-runs/test_external_pyca/../../include -I/home/runner/work/openssl/openssl/test-runs/test_external_pyca/../../include -L/home/runner/work/openssl/openssl/test-runs/test_external_pyca/../.. pytest -n auto tests --wycheproof-root=../wycheproof
````

**Comment by t8m** [comment]
> Has this been broken since it got put in (6 years ago)

AFAIK no, at some point it was definitely linking to the openssl being built.

**Comment by t8m** [comment]
Fix in https://github.com/openssl/openssl/pull/19400

**Comment by alex** [comment]
`-L` should be in `LDFLAGS` I believe.

**Comment by DDvO** [comment]
> It's because of https://github.com/openssl/openssl/blob/openssl-3.0/crypto/pkcs12/p12_kiss.c#L92
> shift was changed to put an error on the stack if you shfit from an enpty stack

Very sorry for my oversight.

> Fix in #19400

Alternative direct fix is in #19414.
